internal/kafka/command_topic_consume.go

```diff
@@ -212,12 +234,135 @@ func (c *command) consume(cmd *cobra.Command, args []string) error {
 		Out:         cmd.OutOrStdout(),
 		Subject:     subject,
 		Properties: ConsumerProperties{
+			Delimiter:   delimiter,
+			FullHeader:  fullHeader,
 			PrintKey:    printKey,
 			PrintOffset: printOffset,
-			FullHeader:  fullHeader,
+			SchemaPath:  schemaPath,
 			Timestamp:   timestamp,
+		},
+	}
+	return RunConsumer(consumer, groupHandler)
+}
+
+func (c *command) consumeOnPrem(cmd *cobra.Command, args []string) error {
+	printKey, err := cmd.Flags().GetBool("print-key")
+	if err != nil {
+		return err
+	}
+
+	printOffset, err := cmd.Flags().GetBool("print-offset")
+	if err != nil {
+		return err
+	}
+
+	fullHeader, err := cmd.Flags().GetBool("full-header")
+	if err != nil {
+		return err
+	}
+
+	timestamp, err := cmd.Flags().GetBool("timestamp")
+	if err != nil {
+		return err
+	}
+
+	delimiter, err := cmd.Flags().GetString("delimiter")
+	if err != nil {
+		return err
+	}
+
+	keyFormat, err := cmd.Flags().GetString("key-format")
+	if err != nil {
+		return err
+	}
+
+	valueFormat, err := cmd.Flags().GetString("value-format")
+	if err != nil {
+		return err
+	}
+
+	configFile, err := cmd.Flags().GetString("config-file")
+	if err != nil {
+		return err
+	}
+	config, err := cmd.Flags().GetStringSlice("config")
+	if err != nil {
+		return err
+	}
+
+	consumer, err := newOnPremConsumer(cmd, c.clientID, configFile, config)
+	if err != nil {
+		return errors.NewErrorWithSuggestions(
+			fmt.Sprintf(errors.FailedToCreateConsumerErrorMsg, err),
+			errors.OnPremConfigGuideSuggestions,
+		)
+	}
+	log.CliLogger.Tracef("Create consumer succeeded")
+
+	if err := c.refreshOAuthBearerToken(cmd, consumer); err != nil {
+		return err
+	}
+
+	adminClient, err := ckafka.NewAdminClientFromConsumer(consumer)
+	if err != nil {
+		return fmt.Errorf(errors.FailedToCreateAdminClientErrorMsg, err)
+	}
+	defer adminClient.Close()
+
+	topicName := args[0]
+	if err := ValidateTopic(adminClient, topicName); err != nil {
+		return err
+	}
+
+	offset, err := GetOffsetWithFallback(cmd)
+	if err != nil {
+		return err
+	}
+
+	partition, err := cmd.Flags().GetInt32("partition")
+	if err != nil {
+		return err
+	}
+	partitionFilter := PartitionFilter{
+		Changed: cmd.Flags().Changed("partition"),
+		Index:   partition,
+	}
+
+	rebalanceCallback := GetRebalanceCallback(offset, partitionFilter)
+	if err := consumer.Subscribe(topicName, rebalanceCallback); err != nil {
+		return err
+	}
+
+	output.ErrPrintln(c.Config.EnableColor, errors.StartingConsumerMsg)
+
+	var srClient *schemaregistry.Client
+	if slices.Contains(serdes.SchemaBasedFormats, valueFormat) {
+		srClient, err = c.GetSchemaRegistryClient(cmd)
+		if err != nil {
+			return err
+		}
+	}
+
+	dir, err := sr.CreateTempDir()
```

Comment:
We should not be importing other `internal` packages. That's a surefire way to get import cycles